src/coreclr/vm/jitinterface.cpp

```diff
@@ -12007,8 +12007,20 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
     if (!field->IsThreadStatic() && pEnclosingMT->IsClassInited() && IsFdInitOnly(field->GetAttributes()))
     {
         GCX_COOP();
-        if (field->IsObjRef())
+        if (field->IsRVA())
```

Comment:
Instead of special-casing IsRVA, can we just add handling of `valueOffset` to the existing path
https://github.com/dotnet/runtime/pull/78783/files#diff-98b4e379205d747e922689cb993f985be190fedb509fd486e99280c842703472R12055-R12058 ?

Reply:
Good point! Should also allow us to query sub-fields of structs now